package/MDAnalysis/analysis/results.py

```diff
@@ -0,0 +1,315 @@
+
+    def __init__(self, *args, **kwargs):
+        kwargs = dict(*args, **kwargs)
+        if "data" in kwargs.keys():
+            raise AttributeError(f"'data' is a protected dictionary attribute")
+        self.__dict__["data"] = {}
+        self.update(kwargs)
+
+    def __setitem__(self, key, item):
+        self._validate_key(key)
+        super().__setitem__(key, item)
+
+    def __setattr__(self, attr, val):
+        if attr == "data":
+            super().__setattr__(attr, val)
+        else:
+            self.__setitem__(attr, val)
+
+    def __getattr__(self, attr):
+        try:
+            return self[attr]
+        except KeyError as err:
+            raise AttributeError(f"'Results' object has no attribute '{attr}'") from err
+
+    def __delattr__(self, attr):
+        try:
+            del self[attr]
+        except KeyError as err:
+            raise AttributeError(f"'Results' object has no attribute '{attr}'") from err
+
+    def __getstate__(self):
+        return self.data
+
+    def __setstate__(self, state):
+        self.data = state
+
+
+class ResultsGroup:
+    """
+    Management and aggregation of results stored in :class:`Results` instances.
+
+    A :class:`ResultsGroup` is an optional description for :class:`Result` "dictionaries"
+    that are used in analysis classes based on :class:`AnalysisBase`. For each *key* in a
+    :class:`Result` it describes how multiple pieces of the data held under the key are
+    to be aggregated. This approach is necessary when parts of a trajectory are analyzed
+    independently (e.g., in parallel) and then need to me merged (with :meth:`merge`) to
+    obtain a complete data set.
+
+    Parameters
+    ----------
+    lookup : dict[str, Callable], optional
+        aggregation functions lookup dict, by default None
+
+    Examples
+    --------
+
+    .. code-block:: python
+
+        from MDAnalysis.analysis.results import ResultsGroup, Results
+        group = ResultsGroup(lookup={'mass': ResultsGroup.float_mean})
+        obj1 = Results(mass=1)
+        obj2 = Results(mass=3)
+        assert {'mass': 2.0} == group.merge([obj1, obj2])
+    
+
+    .. code-block:: python
+
+        # you can also set `None` for those attributes that you want to skip
+        lookup = {'mass': ResultsGroup.float_mean, 'trajectory': None}
+        group = ResultsGroup(lookup)
+        objects = [Results(mass=1, skip=None), Results(mass=3, skip=object)]
+        assert group.merge(objects, require_all_aggregators=False) == {'mass': 2.0}
+
+    .. versionadded:: 2.8.0
+    """
+
+    def __init__(self, lookup: dict[str, Callable] = None):
+        self._lookup = lookup
+
+    def merge(self, objects: Sequence[Results], require_all_aggregators: bool = True) -> Results:
+        """Merge results into a single object. If objects contain single
+           element, returns it while ignoring _lookup attribute.
+
+        Parameters
+        ----------
```

Comment:
document the input Results!
```suggestion
        ----------
        objects : Sequence[Results]
            Multiple :class:`Results` instances with the same data attributes.
```